### PR TITLE
Fix claim.scopes is null cases

### PR DESCRIPTION
**Pull request by zhangxin511**
See following errors when deploying, and fix these cases
image

Tests:
image

### CODE REVIEW

File: server/routerlicious/packages/services-client/src/auth.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export function validateTokenClaims(
 		);
 	}
 
-	if (claims.scopes === undefined || claims.scopes.length === 0) {
```
Comment (connorskees) [inline server/routerlicious/packages/services-client/src/auth.ts:35]: How hard is it to fix the typing of `ITokenClaims`? That would make future issues like this impossible.

Comment (zhangxin511) [reply server/routerlicious/packages/services-client/src/auth.ts:35]: I do not believe add typing would fix issue here.

Typing restrictions is simply a compile time check, and the `null` get from runtime and from a 3rd party library, type restrictions would not change the runtime behavior if my understanding is correct.

Comment (connorskees) [reply server/routerlicious/packages/services-client/src/auth.ts:35]: I mean to say, currently the definition of `ITokenClaims.scope` is typed as `string[]` or `string[] | undefined`. This typing doesn't accurately reflect reality, so the condition above didn't throw a compile-time error, even though the value could potentially be `null`. If the typing were changed to `string[] | null | undefined`, that would prevent a similar error from slipping through.

Comment (zhangxin511) [reply server/routerlicious/packages/services-client/src/auth.ts:35]: Oh I see what you mean, good point.

**Pull request closed without merging**